src/frontend/src/expr/utils.rs

```diff
@@ -486,29 +485,29 @@ impl WatermarkAnalyzer {
     }
 
     fn visit_function_call(&self, func_call: &FunctionCall) -> WatermarkDerivation {
+        use WatermarkDerivation::{Constant, Nondecreasing, Watermark};
         match func_call.func_type() {
             ExprType::Unspecified => unreachable!(),
-            ExprType::Add | ExprType::Multiply => match self.visit_binary_op(func_call.inputs()) {
-                (WatermarkDerivation::Constant, WatermarkDerivation::Constant) => {
-                    WatermarkDerivation::Constant
-                }
-                (WatermarkDerivation::Constant, WatermarkDerivation::Watermark(idx))
-                | (WatermarkDerivation::Watermark(idx), WatermarkDerivation::Constant) => {
-                    WatermarkDerivation::Watermark(idx)
+            ExprType::Add | ExprType::Subtract | ExprType::TumbleStart => {
```

Comment:
should `ExprType::Multiply` be here too?

Reply:
If the constant multiplier is negative, then we can't derive watermark.